packages/components/src/progress-bar/index.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import type { ForwardedRef } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { forwardRef } from '@wordpress/element';
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import * as ProgressBarStyled from './styles';
+import type { ProgressBarProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+
+// Width of the indicator for the indeterminate progress bar
+export const INDETERMINATE_TRACK_WIDTH = 50;
+
+function UnforwardedProgressBar(
+	props: WordPressComponentProps< ProgressBarProps, 'div', false >,
+	ref: ForwardedRef< HTMLProgressElement >
+) {
+	const { className, id: idProp, value } = props;
+	const id = useInstanceId( ProgressBar, 'progress-bar', idProp );
```

Comment:
Nit: We don't really need this `useInstanceId` step because we're not using the id internally in any way. It's the consumer who needs to do a `useInstanceId` if they want a unique custom id. `id` can just be part of the rest props that we forward to `progress`.

Reply:
I think [this is my fault](https://github.com/WordPress/gutenberg/pull/53030#discussion_r1277992619)!

Reply:
No problem at all! I did have some doubts about this and should have questioned it in the first place.

Anyway, resolved as part of f0c3be6d7ca283c5e71d7338675cdd20c542113d